ts/Series/ParetoSeries/ParetoSeries.ts

```diff
@@ -0,0 +1,278 @@
+    );
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<ParetoPoint> = void 0 as any;
+    public points: Array<ParetoPoint> = void 0 as any;
+    public options: ParetoSeriesOptions = void 0 as any;
+
+    public hasDerivedData: Highcharts.DerivedSeries['hasDerivedData'] = void 0 as any;
+    public initialised: Highcharts.DerivedSeries['initialised'] = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /**
+     * Calculate y sum and each percent point.
+     *
+     * @private
+     * @function Highcharts.Series#sumPointsPercents
+     *
+     * @param {Array<number>} yValues
+     * Y values
+     *
+     * @param {Array<number>} xValues
+     * X values
+     *
+     * @param {number} sum
+     * Sum of all y values
+     *
+     * @param {boolean} [isSum]
+     * Declares if calculate sum of all points
+     *
+     * @return {number|Array<number,number>}
+     * Returns sum of points or array of points [x,sum]
+     *
+     * @requires modules/pareto
+     */
+    public sumPointsPercents<T extends(boolean | undefined)>(
+        yValues: Array<number>,
+        xValues: Array<number>,
+        sum: number,
+        isSum?: T
+    ): (T extends true ? number : Array<Array<number>>) {
+        var sumY = 0,
+            sumPercent = 0,
+            percentPoints: Array<Array<number>> = [],
+            percentPoint: (number | undefined);
+
+        yValues.forEach(function (point: number, i: number): void {
+            if (point !== null) {
+                if (isSum) {
+                    sumY += point;
+                } else {
+                    percentPoint = (point / sum) * 100;
+                    percentPoints.push([
+                        xValues[i],
+                        correctFloat(sumPercent + percentPoint)
+                    ]);
+                    sumPercent += percentPoint;
+                }
+            }
+        });
+
+        return (isSum ? sumY : percentPoints) as (
+            T extends true ? number : Array<Array<number>>
+        );
+    }
+}
+
+/* *
+ *
+ *  Prototype properties
+ *
+ * */
+
+interface ParetoSeries {
+    addBaseSeriesEvents: typeof DerivedSeriesMixin['addBaseSeriesEvents'];
+
+    addEvents: typeof DerivedSeriesMixin['addEvents'];
+
+    eventRemovers: Highcharts.DerivedSeries['eventRemovers'];
+
+    init: typeof DerivedSeriesMixin['init'];
+
+    pointClass: typeof ParetoPoint;
+
+    setBaseSeries: typeof DerivedSeriesMixin['setBaseSeries'];
+
+    setDerivedData: Highcharts.DerivedSeries['setDerivedData'];
+}
+
+extend(ParetoSeries.prototype,
+    merge(DerivedSeriesMixin, {
+        /* eslint-disable no-invalid-this, valid-jsdoc */
+        /**
+         * Calculate sum and return percent points.
+         *
+         * @private
+         * @function Highcharts.Series#setDerivedData
+         * @requires modules/pareto
+         */
+        setDerivedData: (function (this: ParetoSeries): void {
```

Comment:
To avoid confusion during future refactoring, convert it to manual mixin apply. See https://github.com/highcharts/highcharts/pull/14719 for procedure.

Reply:
After that you can move the custom implementation setDerivedData to the class pattern.